python-checks/src/main/java/org/sonar/python/checks/NumpyRandomStateCheck.java

```diff
@@ -0,0 +1,104 @@
+import java.util.Optional;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Tree;
+
+@Rule(key = "S6711")
+public class NumpyRandomStateCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "Do not use legacy functions that relies on \"numpy.random.RandomState\", use their \"numpy.random.Generator\" counterpart instead.";
+
+  private static final String LEGACY_MODULE_NAME = "numpy.random.RandomState";
+  private static final List<String> LEGACY_RANDOM_FUNCTIONS = List.of(
+      "numpy.random.beta",
+      "numpy.random.binomial",
+      "numpy.random.bytes",
+      "numpy.random.chisquare",
+      "numpy.random.choice",
+      "numpy.random.dirichlet",
+      "numpy.random.exponential",
+      "numpy.random.f",
+      "numpy.random.gamma",
+      "numpy.random.geometric",
+      "numpy.random.get_state",
+      "numpy.random.gumbel",
+      "numpy.random.hypergeometric",
+      "numpy.random.laplace",
+      "numpy.random.logistic",
+      "numpy.random.lognormal",
+      "numpy.random.logseries",
+      "numpy.random.multinomial",
+      "numpy.random.multivariate_normal",
+      "numpy.random.negative_binomial",
+      "numpy.random.noncentral_chisquare",
+      "numpy.random.noncentral_f",
+      "numpy.random.normal",
+      "numpy.random.pareto",
+      "numpy.random.permutation",
+      "numpy.random.poisson",
+      "numpy.random.power",
+      "numpy.random.rand",
+      "numpy.random.randint",
+      "numpy.random.randn",
+      "numpy.random.random",
+      "numpy.random.random_integers",
+      "numpy.random.random_sample",
+      "numpy.random.ranf",
+      "numpy.random.rayleigh",
+      "numpy.random.sample",
+      "numpy.random.seed",
+      "numpy.random.set_state",
+      "numpy.random.shuffle",
+      "numpy.random.standard_cauchy",
+      "numpy.random.standard_exponential",
+      "numpy.random.standard_gamma",
+      "numpy.random.standard_normal",
+      "numpy.random.standard_t",
+      "numpy.random.triangular",
+      "numpy.random.uniform",
+      "numpy.random.vonmises",
+      "numpy.random.wald",
+      "numpy.random.weibull",
+      "numpy.random.zipf"
+  );
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, NumpyRandomStateCheck::checkNumpyRandomState);
+  }
+
+  private static void checkNumpyRandomState(SubscriptionContext ctx) {
+    CallExpression call = (CallExpression) ctx.syntaxNode();
+    Optional.ofNullable(call.calleeSymbol())
+        .map(Symbol::fullyQualifiedName)
+        .filter(fqn -> fqn.startsWith(LEGACY_MODULE_NAME) || LEGACY_RANDOM_FUNCTIONS.contains(fqn))
+        .ifPresent(symbol -> ctx.addIssue(call.callee(), MESSAGE));
```

Comment:
Very nice and clean way to add the issue! Will definitely make more use of the Java functional interface in the future.